HiP-DataStore/Controllers/ReviewHelper.cs

```diff
@@ -0,0 +1,106 @@
+                return ErrorMessages.CannotAddReviewToContentWithWrongStatus();
+
+            if (reviewIndex.Exists(resourceType.Name, id))
+                return ErrorMessages.ContentAlreadyHasReview(resourceType, id);
+
+            return null;
+        }
+
+        public static string CheckNotFoundGet(int id, ResourceType resourceType, EntityIndex entityIndex, ReviewIndex reviewIndex)
+        {
+            if (!entityIndex.Exists(resourceType, id))
+                return ErrorMessages.ContentNotFound(resourceType, id);
+
+            if (!reviewIndex.Exists(resourceType.Name, id))
+                return ErrorMessages.ReviewNotFound(resourceType, id);
+
+            return null;
+        }
+
+        public static string CheckNotFoundPut(int id, ResourceType resourceType, EntityIndex entityIndex, ReviewIndex reviewIndex)
+        {
+            if (!entityIndex.Exists(resourceType, id))
+                return ErrorMessages.ContentNotFound(resourceType, id);
+
+            if (!reviewIndex.Exists(resourceType.Name, id))
+                return ErrorMessages.ReviewNotFound(resourceType, id);
+
+            return null;
+        }
+
+        public static bool CheckForbidPut(ReviewArgs oldReviewArgs, IIdentity identity, ReviewIndex reviewIndex, int reviewId)
+        {
+            if (!oldReviewArgs.ReviewableByStudents == true && !UserPermissions.IsSupervisorOrAdmin(identity))
+                return true;
+            // Only reviewers, owner and admins/supervisors are allowed to comment
+            if (!UserPermissions.IsAllowedToCommentReview(identity, oldReviewArgs.Reviewers, reviewIndex.Owner(reviewId)))
+                return true;
+
+            return false;
+        }
+
+        public static ReviewArgs UpdateReviewArgs(ReviewArgs args, ReviewArgs oldReviewArgs, IIdentity identity)
+        {
+            // Only admins and supervisors are permitted to change the amount of students that are necessary to approve a review
+            if (args.StudentsToApprove != oldReviewArgs.StudentsToApprove && !UserPermissions.IsSupervisorOrAdmin(identity))
+                args.StudentsToApprove = oldReviewArgs.StudentsToApprove;
+            // Only admins and supervisors are permitted to change whether students are allowed to approve a review
+            if (args.ReviewableByStudents != oldReviewArgs.ReviewableByStudents && !UserPermissions.IsSupervisorOrAdmin(identity))
+                args.ReviewableByStudents = oldReviewArgs.ReviewableByStudents;
+
+            args.EntityType = oldReviewArgs.EntityType;
+            args.EntityId = oldReviewArgs.EntityId;
+            args.Comments = oldReviewArgs.Comments;
+            args.Approved = oldReviewArgs.Approved;
+
+            if (args.ReviewableByStudents == false)
+                args.StudentsToApprove = 0;
+
+            // keep old values if no new ones are specified
+            if (args.Description == null)
```

Comment:
I think this in inconsistent to our current behavior. By our current API design (e.g. for Exhibits) we assume that always the whole object is provided (including old values). This means that the copying of the old values is unnecessary.